### PR TITLE
Feature/cloud 1412 coverlet msbuild

### PR DESCRIPTION
**Pull request by naveen-variant**
# Description

Adding coverlet msbuild

Fixes [#23](https://usxtech.atlassian.net/browse/LAZY-1412)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

```yaml
Pre reqs:
  - Create branch on any dotnet project by pointing to dotnet action ms build branch
Test Case 1 [Run on repos with new dotnet action changes]:
- Steps
  - Can re-run the branches in the sample runs below and see it is running new nuget related changes.
  - Or  create a new branch on  new reposand see new nuget related changes are running. 
  - Pick projects where nuget_push_enabled=true on those projects as this nuget related change.
- Sample runs
  - https://github.com/variant-inc/trailers-api/actions/runs/1713173809
  - https://github.com/variant-inc/schedule-adherence/runs/4854105773?check_suite_focus=true
```

```text
Note: I will update the action.yaml to minor version as i receive the approvals on this PR.
```

- [ ] Unit Tests
- [x] Sanity Testing

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by vibindaniel** [comment]
@naveen-variant How did you create image with tag test-latest? The workflow for branches uses git sha.

File: scripts/cover.ps1 [hunk 1]
```diff
@@ -28,7 +28,9 @@ $tests = $(dotnet sln $solutionFileDir list) | Select-Object -Skip 2 | Where-Obj
 $tests | ForEach-Object {
   $file = [System.IO.DirectoryInfo]"$_"
   $parent = $($file.parent.fullname)
+  ce dotnet add $parent package coverlet.msbuild
   ce dotnet add $parent package coverlet.collector
+
```
Comment (vibindaniel) [inline scripts/cover.ps1:33]: remove this line

File: action.yml [hunk 2]
```diff
@@ -37,6 +37,6 @@ inputs:
 
 runs:
   using: "docker"
-  image: docker://public.ecr.aws/p4v7w0a5/lazy/dotnet-action:v1.1.4
+  image: docker://public.ecr.aws/p4v7w0a5/lazy/dotnet-action:test-latest
```
Comment (vibindaniel) [inline action.yml:40]: change to next minor version 1.2.0

**Review by LeeHughsVariant — APPROVED** [review]
Tested and passed.
Aside from Vibin's comments, LGTM 👍

File: scripts/cover.ps1 (hunk 1 above)
Comment (naveen-variant) [reply scripts/cover.ps1:33]: ok.

File: action.yml (hunk 2 above)
Comment (naveen-variant) [reply action.yml:40]: > change to next minor version 1.2.0

Sure

File: scripts/cover.ps1 (hunk 1 above)
Comment (naveen-variant) [reply scripts/cover.ps1]: Removed empty line though.

**Comment by rchengvariant** [comment]
@naveen-variant 

Sanity Test: https://github.com/variant-inc/demo-app/runs/4858854471?check_suite_focus=true

Looks good. Like Vibin said, dont forget to change to 1.2.0. Will approve after that.

**Review by rchengvariant — APPROVED** [review]

**Pull request merged**